pkg/genericapiserver/config_selfclient.go

```diff
@@ -75,24 +75,46 @@ func (s *SecureServingInfo) NewSelfClientConfig(token string) (*restclient.Confi
 		BearerToken: token,
 	}
 
-	// find certificate for host: either explicitly given, from the server cert bundle or one of the SNI certs
+	// find certificate for host: either explicitly given, from the server cert bundle or one of the SNI certs,
+	// but only return CA:TRUE certificates.
 	var derCA []byte
 	if s.CACert != nil {
```

Comment:
if the user specifies `--tls-ca-file`, it can contain multiple certs and we should trust all of them when connecting locally